pkg/globalnet/controllers/global_ingressip_controller_test.go

```diff
@@ -82,32 +79,94 @@ var _ = Describe("GlobalIngressIP controller", func() {
 	}
 
 	When("a GlobalIngressIP for a cluster IP Service is created", func() {
-		testGlobalIngressIPCreated(t, clusterIPServiceIngress, t.awaitServiceIngressRules, t.awaitNoServiceIngressRules,
-			kubeProxyIPTableChainName)
+		testGlobalIngressIPCreatedClusterIPSvc(t, clusterIPServiceIngress)
 	})
 
 	When("a GlobalIngressIP for a headless Service is created", func() {
-		testGlobalIngressIPCreated(t, headlessServiceIngress, awaitHeadlessServicePodRules, awaitNoHeadlessServicePodRules, podIP)
+		testGlobalIngressIPCreatedHeadlessSvc(t, headlessServiceIngress, awaitHeadlessServicePodRules, awaitNoHeadlessServicePodRules, podIP)
 	})
 
 	When("a GlobalIngressIP for a cluster IP Service exists on startup", func() {
-		testExistingGlobalIngressIP(t, clusterIPServiceIngress, t.awaitServiceIngressRules)
+		testExistingGlobalIngressIPClusterIPSvc(t, clusterIPServiceIngress)
 	})
 
 	When("a GlobalIngressIP for a headless Service exists on startup", func() {
-		testExistingGlobalIngressIP(t, headlessServiceIngress, awaitHeadlessServicePodRules)
+		testExistingGlobalIngressIPHeadlessSvc(t, headlessServiceIngress, awaitHeadlessServicePodRules)
 	})
 })
 
-func testGlobalIngressIPCreated(t *globalIngressIPControllerTestDriver, ingressIP *submarinerv1.GlobalIngressIP,
+func testGlobalIngressIPCreatedClusterIPSvc(t *globalIngressIPControllerTestDriver, ingressIP *submarinerv1.GlobalIngressIP) {
+	JustBeforeEach(func() {
+		service := newClusterIPService()
+		t.createService(service)
+		t.createGlobalIngressIP(ingressIP)
+	})
+
+	It("should successfully allocate a global IP", func() {
+		t.awaitIngressIPStatusAllocated(globalIngressIPName)
+		allocatedIP := t.getGlobalIngressIPStatus(globalIngressIPName).AllocatedIP
+		Expect(allocatedIP).ToNot(BeEmpty())
+	})
+
+	It("should successfully create an internal submariner service", func() {
+		internalSvcName := controllers.GetInternalSvcName(serviceName)
+		intSvc := t.awaitService(internalSvcName)
+		externalIP := intSvc.Spec.ExternalIPs[0]
+		Expect(externalIP).ToNot(BeEmpty())
+		finalizer := intSvc.GetFinalizers()[0]
+		Expect(finalizer).To(Equal(controllers.InternalServiceFinalizer))
+		gIP := intSvc.GetAnnotations()[controllers.GlobalIngressIP]
+		Expect(gIP).To(Equal(externalIP))
```

Comment:
To improve readability please use newlines between logical sections.

```suggestion
	        intSvc := t.awaitService(controllers.GetInternalSvcName(serviceName))
                externalIP := intSvc.Spec.ExternalIPs[0]
                Expect(externalIP).ToNot(BeEmpty())
		 
                 Expect(intSvc.GetFinalizers).To(Equal([]string{controllers.InternalServiceFinalizer}))

                 gIP := intSvc.GetAnnotations()[controllers.GlobalIngressIP]
                 Expect(gIP).To(Equal(externalIP))
```